src/DatabaseProvider/index.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+import React from 'react'
+import Database from '../Database'
+
+const { Provider, Consumer } = React.createContext({})
```

Comment:
@kokusGr What would the type of the defaultValue? After removing the default value, flow 

```bash
Missing type annotation for T. T is a type parameter declared in function type [1] and was implicitly instantiated at
call of method createContext [2].

     src/DatabaseProvider/index.js
       2│ import React from 'react'
       3│ import Database from '../Database'
       4│
 [2]   5│ const { Provider, Consumer } = React.createContext()
       6│
       7│ export type Props = {
       8│   database: Database,

     /private/tmp/flow/flowlib_2a8a6d42/react.js
 [1] 223│   declare export function createContext<T>(
     224│     defaultValue: T,
     225│   ): React$Context<T>;
```

Should it be explicitly undefined?
```javascript
React.createContext<undefined>()
```

Reply:
Hmm i didn't get an error, but it's still wasn't accurate solution. The problem is that we don't have any `defaultValue` so proper type should be `React.createContext<?Database>()`, but this complicates consuming db. So we can:

1. Cheat - `const { Provider, Consumer) = (React.createContext(): any)`. We know that db will always be there, because `DatabaseProvider` would throw an error so we are fairly safe.
2. Treat `Database` as nullable which is a safer but more verbose and kinda redundant only to make Flow happy.

I think we can go with nr 1 :) WDYT @Aidurber @mobily ?